tests_dirac/test_it.py

```diff
@@ -206,3 +206,43 @@ async def test_states_and_cancel(tmp_path: Path) -> None:  # noqa: WPS217 readab
         await fs.delete(gdescription)
         await fs.close()
         await scheduler.close()
+
+
+@pytest.mark.anyio
+async def test_failing_job(  # noqa: WPS217 single piece of code for readablilty
+    tmp_path: Path,
+) -> None:
+    fs_config = DiracFileSystemConfig(
+        lfn_root="/tutoVO/user/c/ciuser/bartenderjobs",
+        storage_element="StorageElementOne",
+    )
+    fs = DiracFileSystem(fs_config)
+    sched_config = DiracSchedulerConfig(storage_element=fs_config.storage_element)
+    scheduler = DiracScheduler(sched_config)
+    # emulate external job id with job1 subdir
+    # job_dir.name is used as directory to upload inputfiles to lfn_root
+    job_dir = tmp_path / "job4"
+    job_dir.mkdir()
+    description = JobDescription(
+        command="echo icannotwork && ls /idonotexist",
+        job_dir=job_dir,
+    )
+    gdescription = fs.localize_description(description, tmp_path)
+    try:
+        await fs.upload(description, gdescription)
+
+        job_id = await scheduler.submit(gdescription)
+
+        assert job_id
+
+        await wait_for_job(scheduler, job_id, expected="error")
+
+        stdout, stderr = await scheduler.logs(job_id, description.job_dir)
+
+        assert "icannotwork" in stdout
+        assert "idonotexist" in stderr
```

Comment:
Downloading output will fail due to output.tar not being uploaded to grid storage. Check if failure is handled gracefully.

Reply:
Calling download() in test